### ISSUE TITLE
Выбор лицензии на разрабатываемое ПО, документацию, Wiki

### ISSUE THREAD

**Issue by easimonenko** [severe-island/navee#2]
Важный вопрос выбора лицензии на разрабатываемое ПО. Моё предпочтение такое: семейство лицензий GNU GPL 3 или новее. Иногда выбор лицензии обусловлен лицензией на базовое ПО. Например, если там принято всё выпускать под MIT, то значит и нам скорее всего лучше взять за основу MIT.

Для части кода может быть применима CC-0 1.0 (Public Domain). Обычно так поступают для примеров кода.

Для документации и Wiki можно применять GNU FDL 3 или CC-BY-SA 4.

**Comment by razzeeyy** [comment]
Кстати да, как-то я забыл за этот момент, как за само собой разумеющееся.

Я на данный момент склоняюсь больше к выбору в пользу MIT лицензии для проекта и кода.
Т.к. обычно проекты с MIT лицензией получают большее распространение, и возможность использовать их даже в проприетарном обеспечении позволяет получать неплохие патчи и контрибуции, если такое ПО используется у крупных игроков рынка.

GNU GPL 3 хорошая свободная лицензия, но иногда вот эта свободность и копилефт может таки перегнуть палку. В частности например, ну написал человек (или небольшая компания) небольшую модификацию чисто для своих нужд, что же им теперь выкладывать исходный код? Даже если это просто сдвиг какой-нибудь кнопки в GUI туда-сюда, или другое изменение которое практически мало кому будет интересно для включения в основной проект.

Про лицензии на документации, мне надо немного изучить момент, чем одна от другого отличается и составить своё предпочтение в этом. Отпишусь когда прошарюсь (где-то в районе ближайшего дня).

**Comment by razzeeyy** [comment]
В пользу лицензии для документации я склоняюсь к выбору CC-BY-SA 4.

Как можно видеть из краткого сравнения: https://en.wikipedia.org/wiki/Wikipedia:Comparison_of_GFDL_and_CC_BY-SA
У GNU опять геморрой из-за копилефта:
1) при использовании материалов нужно прилагать ссылку на оригинал лицензии,
2) также лицензия позволяет иметь немодифицируемые разделы
Считаю первый пункт вообще не юзер френдли, хотя он и не так релевантен для документации, а второй пункт уже открывает потенциальный простор для троллинга и создания неизменяемых разделов на каждый чих.

Вообщем и целом в моих глазах выигрывает CC-BY-SA 4, к тому же она имеет красивое и человекопонятное краткое описание.
https://creativecommons.org/licenses/by-sa/4.0/

**Comment by easimonenko** [comment]
В своих произведениях я использую либо CC-BY-SA, либо CC-BY-ND. Поэтому поддерживаю твоё мнение по выбору лицензии CC-BY-SA 4.

Что касается лицензии на исходные тексты, предлагаю отложить этот вопрос до написания кода, когда будет вырисовываться программное окружение, в котором будет этот код разрабатываться и работать. Под это окружение лицензия тоже должна подходить.

**Comment by razzeeyy** [comment]
Да, может быть так выйдет, что нам будет необходимо опереться на какую-то GPL библиотеку...
Хорошо подмечено.
